src/Orckestra.Composer.Search/Services/SearchViewService.cs

```diff
@@ -99,14 +99,32 @@ public virtual async Task<SearchViewModel> GetSearchViewModelAsync(SearchCriteri
                     viewModel.FacetSettings.SelectedFacets,
                     viewModel.ProductSearchResults.CategoryFacetCounts).ConfigureAwait(false);
 
-                // Json context for Facets
-                viewModel.FacetSettings.Context["CategoryFacetValuesTree"] = viewModel.FacetSettings.CategoryFacetValuesTree;
-                viewModel.FacetSettings.Context["SelectedFacets"] = viewModel.FacetSettings.SelectedFacets;
-                viewModel.FacetSettings.Context["Facets"] = viewModel.ProductSearchResults.Facets.Where(f => !f.FieldName.StartsWith(SearchConfiguration.CategoryFacetFiledNamePrefix));
-                viewModel.FacetSettings.Context["PromotedFacetValues"] = viewModel.ProductSearchResults.PromotedFacetValues;
+
+                var selectedFacetValues = new HashSet<string>(
+                    viewModel.FacetSettings?.CategoryFacetValuesTree?.ChildNodes
+                        ?.SelectMany(GetSelectedFacetValues) ?? Array.Empty<string>());
+
+                if (viewModel.FacetSettings != null)
+                {
+                    viewModel.FacetSettings.SelectedFacets.Facets = viewModel.FacetSettings.SelectedFacets.Facets
+                        .Where(item =>
+                            selectedFacetValues.Contains(item.Value)
+                            || !item.FieldName.StartsWith(SearchConfiguration.CategoryFacetFiledNamePrefix))
+                        .ToList();
+
+                    // Json context for Facets
+                    viewModel.FacetSettings.Context["CategoryFacetValuesTree"] =
+                        viewModel.FacetSettings.CategoryFacetValuesTree;
+                    viewModel.FacetSettings.Context["SelectedFacets"] = viewModel.FacetSettings.SelectedFacets;
+                    viewModel.FacetSettings.Context["Facets"] = viewModel.ProductSearchResults.Facets.Where(f =>
+                        !f.FieldName.StartsWith(SearchConfiguration.CategoryFacetFiledNamePrefix));
+                    viewModel.FacetSettings.Context["PromotedFacetValues"] =
+                        viewModel.ProductSearchResults.PromotedFacetValues;
+                }
```

Comment:
move this block (// Json context for Facets) as it was before as viewModel.FacetSettings will be never null
but SelectedFacets can be null